File: 백준/Gold/1931. 회의실 배정/회의실 배정.py
from sys import stdin
input = stdin.readline

n = int(input().rstrip())

time = [list(map(int, input().split())) for _ in range(n)]
time_sorted = sorted(time, key=lambda x: (x[1], x[0]))

answer = 0
current_end = 0

for start, end in time_sorted:
    if start >= current_end:
        answer += 1
        current_end = end

print(answer)

File: 백준/Gold/1931. 회의실 배정/회의실 배정.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

const input = [];
rl.on("line", (line) => {
  input.push(line.trim().split(" ").map(Number));
  if (input.length > 0) {
    const n = input[0][0];

    if (input.length === n + 1) rl.close();
  }
}).on("close", () => {
  const n = input[0][0];
  const arr = input.slice(1);
  arr.sort((a, b) => {
    if (a[1] === b[1]) return a[0] - b[0]; // 시작 시간 기준 오름차순
    return a[1] - b[1]; // 끝나는 시간 기준 오름차순
  });

  let count = 0;
  let endTime = 0;

  for (let i = 0; i < n; i++) {
    if (arr[i][0] >= endTime) {
      count++;
      endTime = arr[i][1];
    }
  }

  console.log(count);

  process.exit();
});
